lib/chef/resource/powershell_package_source.rb

```diff
@@ -24,36 +25,131 @@ class PowershellPackageSource < Chef::Resource
+          source_location              "https://pkgs.dev.azure.com/some-org/some-project/_packaging/some_feed/nuget/v2"
+          publish_location             "https://pkgs.dev.azure.com/some-org/some-project/_packaging/some_feed/nuget/v2"
+          trusted                      true
+          provider_name                "chocolatey"
+          action                       :register
+        end
+        ```
+
+        **Add a new PowerShell Script source that is trusted**:
+
+        ```ruby
+        powershell_package_source 'MyDodgyScript' do
+          source_name                  "MyDodgyScript"
+          script_source_location       "https://pkgs.dev.azure.com/some-org/some-project/_packaging/some_feed/nuget/v2"
+          script_publish_location      "https://pkgs.dev.azure.com/some-org/some-project/_packaging/some_feed/nuget/v2"
+          trusted                      true
+          action                       :register
+        end
+        ```
+
+        **Update my existing PSRepository to make it Trusted after all**:
+
+        ```ruby
+        powershell_package_source 'MyPSModule' do
+          source_name                  "MyPSModule"
+          trusted                      true
+          action                       :set
+        end
+        ```
+
+        **Update a Nuget package source with a new name and make it trusted**:
+
+        ```ruby
+        powershell_package_source 'PowerShellModules -> GoldFishBowl' do
+          source_name                  "PowerShellModules"
+          new_name                     "GoldFishBowl"
+          provider_name                "Nuget"
+          trusted                      true
+          action                       :set
+        end
+        ```
+
+        **Update a Nuget package source with a new name when the source is secured with a username and password**:
+
+        ```ruby
+        powershell_package_source 'PowerShellModules -> GoldFishBowl' do
+          source_name                  "PowerShellModules"
+          new_name                     "GoldFishBowl"
+          trusted                      true
+          user_name                    "user@domain.io"
+          user_pass                    "some_secret_password"
+          action                       :set
+        end
+        ```
+
+        **Unregister a package source**:
+
+        ```ruby
+        powershell_package_source 'PowerShellModules' do
+          source_name                  "PowerShellModules"
+          action                       :unregister
+        end
+        ```
+      DOC
 
       property :source_name, String,
-        description: "The name of the package source.",
+        description: "A label that names your package source.",
         name_property: true
 
-      property :url, String,
-        description: "The URL to the package source.",
-        required: [:register]
+      property :new_name, String,
+        description: "Used when updating the name of a NON-PSRepository"
```

Comment:
This description doesn't really help a user understand what this does.

Reply:
Updated it to be clearer